### DISCUSSION TITLE
Metrics reporting integrations and per-endpoint metrics

### DISCUSSION THREAD

**Discussion by tchambers97** [artilleryio/artillery#1626] [Artillery]
Hello, we are testing GraphQL endpoints, where there is one URL and we differentiate the requests in the metrics with the name and the metrics-by-endpoint plugin. Do any of the current metrics reporting integrations send data by request name so we can track these between test runs? I tried the Honeycomb integration, and everything was lumped under the one URL.

**Comment by hassy** [comment]
hi @tchambers97 👋 

Do you have an example of a request spec that does not work as expected?

Is the issue that a `name` property is set on a request, but the URL is used instead for reporting request metrics in Honeycomb - is that correct?

**Reply by tchambers97** [reply]
Yes, that is the issue. The URL is the same for all API requests, so the way we distinguish them is by name. I researched a little further, and I see in `honeycomb.js` where the payload is url, host, method, and statusCode. I think adding `name` to this list would make the integration more useful for us.
We're also looking into the StatsD integration, which includes all of the metrics we collect without any changes, but we will need our cloud engineering team to implement the New Relic/StatsD integration in order for that to be useful for us.
